include/linker/sections.h

```diff
@@ -46,7 +46,15 @@
 
 /* Interrupts */
 #define IRQ_VECTOR_TABLE	.gnu.linkonce.irq_vector_table
-#define SW_ISR_TABLE		.gnu.linkonce.sw_isr_table
+
+/* Place the table in either RAM or ROM depending on whether we allow
+ * IRQs to be configured at runtime
```

Comment:
I couldn't quite figure out what .gnu.linkonce does and how .gnu.linkonce.d and .gnu.linkonce.r differ from each other. May be a little more verbosity is required here.

Reply:
Use of this has been historical for some time, but when I looked it seems it is undocumented in the 'ld' documentation. I had to go to the source code for binutils:

```
  /* As a GNU extension, if the name begins with .gnu.linkonce, we
     only link a single copy of the section.  This is used to support
     g++.  g++ will emit each template expansion in its own section.
     The symbols will be defined as weak, so that multiple definitions
     are permitted.  The GNU linker extension is to actually discard
     all but one of the sections.  */
```

I'm going to see if I can do the same thing with __attribute__((weak)) which is at least documented.

> how .gnu.linkonce.d and .gnu.linkonce.r differ from each other

These are just sections in the linker script, placed in data and rodata respectively.

Reply:
yeah I think I can do this with __weak. updated PR to follow.